### ISSUE TITLE
Add support for pre-trained reward models

### ISSUE THREAD

**Issue by diegofiori** [nebuly-ai/nebullvm#226]
# Description

OpenAssistant has released on HF the reward models they trained on the open-source datasets. Even if they are not tailored for the user need, we could lavarege them as a starting point for fine-tuning the user reward models.

Available reward models:

- [OpenAssistant/reward-model-deberta-v3-large-v2 · Hugging Face](https://huggingface.co/OpenAssistant/reward-model-deberta-v3-large-v2)
- [OpenAssistant/reward-model-deberta-v3-large · Hugging Face](https://huggingface.co/OpenAssistant/reward-model-deberta-v3-large)
- [OpenAssistant/reward-model-deberta-v3-base · Hugging Face](https://huggingface.co/OpenAssistant/reward-model-deberta-v3-base)
- [OpenAssistant/reward-model-electra-large-discriminator · Hugging Face](https://huggingface.co/OpenAssistant/reward-model-electra-large-discriminator)

# TODO

- [ ]  Add the possibility to load the pretrained checkpoints for the reward model in ChatLLaMA
- [ ]  Write unittests

**Comment by gagan3012** [comment]
Can I work on this?

**Comment by PierpaoloSorbellini** [comment]
Please go ahead, let me know if you need any support or if you have any questions. 
I assigned you to this issue. 
Thank you! @gagan3012
